test/transforms/test_replicate.py

```diff
@@ -0,0 +1,262 @@
+    task_id = "fwp41cUkRmara7CD6l2U3A"
+    prefix = "kind-a"
+    task = {
+        "name": prefix,
+        "replicate": {
+            "target": [
+                task_id,
+            ]
+        },
+    }
+    task_def = get_target_defs()[0]
+    expected = get_expected(prefix, task_def)[0]
+
+    responses.get(f"{TC_ROOT_URL}/api/queue/v1/task/{task_id}", json=task_def)
+    responses.get(
+        f"{TC_ROOT_URL}/api/queue/v1/task/{task_id}/artifacts/public/task-graph.json",
+        body=Exception("Artifact not found!"),
+    )
+
+    result = run_replicate(task)
+    assert len(result) == 1
+    assert result[0] == expected
+
+
+def test_normal_index_path(responses, get_target_defs, get_expected, run_replicate):
+    prefix = "kind-a"
+    task_id = "def"
+    index_path = "foo.bar"
+    task = {
+        "name": prefix,
+        "replicate": {"target": [index_path]},
+    }
+    task_def = get_target_defs()[0]
+    expected = get_expected(prefix, task_def)[0]
+
+    responses.get(
+        f"{TC_ROOT_URL}/api/index/v1/task/{index_path}", json={"taskId": task_id}
+    )
+    responses.get(
+        f"{TC_ROOT_URL}/api/queue/v1/task/{task_id}/artifacts/public/task-graph.json",
+        body=Exception("Artifact not found!"),
+    )
+    responses.get(f"{TC_ROOT_URL}/api/queue/v1/task/{index_path}", json=task_def)
+
+    result = run_replicate(task)
+    assert len(result) == 1
+    assert result[0] == expected
+
+
+def test_decision_task(responses, get_target_defs, get_expected, run_replicate):
+    prefix = "kind-a"
+    task_id = "fwp41cUkRmara7CD6l2U3A"
+    task = {
+        "name": prefix,
+        "replicate": {
+            "target": [
+                task_id,
+            ]
+        },
+    }
+    task_defs = get_target_defs({}, {"task": {"metadata": {"name": "task-c"}}})
+    expected = get_expected(prefix, *task_defs)
+
+    counter = count()
+    responses.get(
+        f"{TC_ROOT_URL}/api/queue/v1/task/{task_id}/artifacts/public/task-graph.json",
+        json={next(counter): task_def for task_def in task_defs},
+    )
+    result = run_replicate(task)
+    assert result == expected
+
+
+@pytest.mark.parametrize(
+    "target_def",
+    (
+        pytest.param(
+            {
+                "attributes": {"foo": "bar"},
+                "task": {"provisionerId": "releng-hardware"},
+            },
+            id="releng-hardware",
+        ),
+        pytest.param(
+            {
+                "attributes": {"foo": "bar"},
+                "task": {"payload": {"features": {"runAsAdministrator": True}}},
+            },
+            id="runAsAdministrator",
+        ),
+        pytest.param(
+            {},  # doesn't match 'include-attrs'
+            id="include-attrs",
+        ),
+        pytest.param(
+            {"attributes": {"foo": "bar", "baz": "1"}},
+            id="exclude-attrs",
+        ),
+    ),
+)
+def test_filters(responses, get_target_defs, run_replicate, target_def):
```

Comment:
nit: `test_filtered_out` might be better here; the first two examples aren't obviously things that should be filtered away based on the IDs.